pkg/backup/backuper_test.go

```diff
@@ -0,0 +1,184 @@
+package backup
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"testing"
+	"time"
+
+	_ "github.com/mattn/go-sqlite3"
+	"github.com/stretchr/testify/require"
+)
+
+func TestBackuperDefault(t *testing.T) {
+	// substitutes the open function to a mocked version
+	open = func(uri string) (DB, error) {
+		db, err := openDatabase(uri)
+		require.NoError(t, err)
+
+		return &tempDatabase{db}, nil
+	}
+
+	// substitutes the newBackupFile function to a mocked version
+	newBackupFile = func(dir string, _ time.Time) (string, time.Time, error) {
+		timestamp := time.Date(2009, 11, 17, 20, 34, 58, 651387237, time.UTC)
+		return createBackupFile(dir, timestamp)
+	}
+
+	controlDB := CreateControlDatabase(t)
```

Comment:
We make use of a control database as ground truth. Check `control.sql`. 

That file creates a table, inserts 1000 rows and remove 500 odd rows to create space for vacuum. SQLite should execute `control.sql` deterministically, which means we can test the db size before and after vacuum.